1. Найти с помощью NumPy SVD для матрицы

$$\begin{pmatrix}
1 & 2 & 0\\ 
0 & 0 & 5\\ 
3 & -4 & 2\\ 
1 & 6 & 5\\ 
0 & 1 & 0
\end{pmatrix}.$$




In [3]:
import numpy as np
np.set_printoptions(precision=3, suppress=True)

A = np.array([[1,2,0],[0,0,5],[3,4,-2],[1,6,5],[0,1,0]])

U, s, W = np.linalg.svd(A)

# Транспонируем матрицу W
V = W.T

# s - список диагональных элементов, его нужно привести к виду диагональной матрицы для наглядности
D = np.zeros_like(A, dtype=float)
D[np.diag_indices(min(A.shape))] = s

print(f'Матрица U:\n{U}')
print(f'Матрица D:\n{D}')
print(f'Матрица V:\n{V}')

Матрица U:
[[ 0.192 -0.226  0.071 -0.902 -0.307]
 [ 0.345  0.606  0.678 -0.097  0.211]
 [ 0.269 -0.756  0.526  0.237  0.152]
 [ 0.874  0.051 -0.417  0.191 -0.15 ]
 [ 0.084 -0.082 -0.292 -0.292  0.903]]
Матрица D:
[[8.974 0.    0.   ]
 [0.    6.292 0.   ]
 [0.    0.    1.372]
 [0.    0.    0.   ]
 [0.    0.    0.   ]]
Матрица V:
[[ 0.209 -0.389  0.897]
 [ 0.757 -0.517 -0.4  ]
 [ 0.62   0.763  0.186]]


2. Для матрицы из предыдущего задания найти:

    а) евклидову норму;
    
    б) норму Фробениуса.

**Решение**

а) $\|A\|_E = \mu_1 = 8.974$

б) $\|A\|_E = \sqrt{\mu_1^2 + \mu_2^2 + \mu_3^2} = 11.046$